src/servers/trigger.ts

```diff
@@ -64,7 +64,9 @@ export async function onchainTransactionEventHandler(
       "payment completed",
     )
 
-    const walletId = await LedgerService().getWalletIdByTransactionHash(tx.id)
+    // FIXME a tx.id should return a walletId[] instead, in case
+    // multiple UXTO goes to the wallet within the same tx
+    const walletId = await LedgerService().getWalletIdByTransactionHash(txHash)
```

Comment:
we shouldn't be using the txHash as the id for onchain transactions - rather txOutPoint.

Reply:
txOutPoint would mean there is an index? we don't have this here (maybe we should), we're really only storing the txHash. btw we are still storing addresses in an array. Is that the intention? I thought this was removed.

Reply:
the addresses are in an array for backwards compatibility but there is never more than 1.

`TxOutPoint` is the way `UTXO`s are referred to in bitcoin. They are a combination of the `txHash` and the index of the specific `UTXO` within the `tx`. 
It is also the bases for the channel ID in lightning (which is the hashed `txOutPoint`).

Reply:
my point is there is no index in this string. so it's not a `TxOutPoint` but really a `TxHash`

Reply:
Right and my point is our logic is flawed / outdated we should be looping through all `UTXO`s in the tx and calling `LedgerService().getWalletIdByTxOutPoint(utxo)` for each.
Just something to be aware of.

Reply:
Or returning `WalletId[]` would also work the way we're doing it now.